bundles/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/influx2/InfluxDB2RepositoryImpl.java

```diff
@@ -0,0 +1,233 @@
+        } else {
+            optionsBuilder.authenticate(configuration.getUser(), configuration.getPassword().toCharArray());
+        }
+        InfluxDBClientOptions clientOptions = optionsBuilder.build();
+
+        final InfluxDBClient createdClient = InfluxDBClientFactory.create(clientOptions);
+        this.client = createdClient;
+        logger.debug("Succesfully connected to InfluxDB. Instance ready={}", createdClient.ready());
+        queryAPI = createdClient.getQueryApi();
+        writeAPI = createdClient.getWriteApi();
+        return checkConnectionStatus();
+    }
+
+    /**
+     * Disconnect from InfluxDB server
+     */
+    @Override
+    public void disconnect() {
+        final InfluxDBClient currentClient = this.client;
+        if (currentClient != null) {
+            currentClient.close();
+        }
+        this.client = null;
+    }
+
+    /**
+     * Check if connection is currently ready
+     *
+     * @return True if its ready, otherwise false
+     */
+    @Override
+    public boolean checkConnectionStatus() {
+        final InfluxDBClient currentClient = client;
+        if (currentClient != null) {
+            Ready ready = currentClient.ready();
+            boolean isUp = ready != null && ready.getStatus() == Ready.StatusEnum.READY;
+            if (isUp) {
+                logger.debug("database status is OK");
+            } else {
+                logger.warn("database not ready");
+            }
+            return isUp;
+        } else {
+            logger.warn("checkConnection: database is not connected");
+            return false;
+        }
+    }
+
+    /**
+     * Write point to database
+     *
+     * @param point
+     */
+    @Override
+    public void write(InfluxPoint point) {
+        final WriteApi currentWriteAPI = writeAPI;
+        if (currentWriteAPI != null) {
+            currentWriteAPI.writePoint(convertPointToClientFormat(point));
+        } else {
+            logger.warn("Write point {} ignored due to writeAPI isn't present", point);
+        }
+    }
+
+    private Point convertPointToClientFormat(InfluxPoint point) {
+        Point clientPoint = Point.measurement(point.getMeasurementName()).time(point.getTime(), WritePrecision.MS);
+        setPointValue(point.getValue(), clientPoint);
+        point.getTags().entrySet().forEach(e -> clientPoint.addTag(e.getKey(), e.getValue()));
+        return clientPoint;
+    }
+
+    private void setPointValue(@Nullable Object value, Point point) {
+        if (value instanceof String) {
+            point.addField(FIELD_VALUE_NAME, (String) value);
+        } else if (value instanceof Number) {
+            point.addField(FIELD_VALUE_NAME, (Number) value);
+        } else if (value instanceof Boolean) {
+            point.addField(FIELD_VALUE_NAME, (Boolean) value);
+        } else if (value == null) {
+            point.addField(FIELD_VALUE_NAME, (String) null);
+        } else {
```

Comment:
Is date type supported?

Reply:
No:
![image](https://user-images.githubusercontent.com/920260/78526692-1578c180-77db-11ea-9010-671ab46d10ec.png)
InfluxDB it's a time series database, and time it's always PK